File: frontend/src/interfaces/UserGame.ts
// src/interfaces/UserGame.ts
import type { User } from "./User";
import type { Game } from "./Game";

export interface UserGame {
  ID: number;
  status: string;         // "active" | "revoked" | ...
  granted_at: string;     // ISO date string
  revoked_at?: string | null;
  game_id: number;
  game?: Game;
  user_id: number;
  user?: User;
}

export interface CreateUserGameRequest {
  status: string;
  granted_at: string;
  revoked_at?: string | null;
  game_id: number;
  user_id: number;
}

export interface UpdateUserGameRequest {
  ID: number;
  status?: string;
  granted_at?: string;
  revoked_at?: string | null;
  game_id?: number;
  user_id?: number;
}

File: backend/entity/user_game.go
package entity

import (
	"time"

	"gorm.io/gorm"
)


type UserGame struct {
	gorm.Model
	Status    string     `json:"status"`      // e.g. "active", "revoked"
	GrantedAt time.Time  `json:"granted_at"`
	RevokedAt *time.Time `json:"revoked_at"`  // เป็น nil ได้ถ้ายังไม่ถูกยกเลิก
	GameID    uint       `json:"game_id"`
	Game      *Game      `gorm:"foreignKey:GameID" json:"game"`
	UserID    uint       `json:"user_id"`
	User      *User      `gorm:"foreignKey:UserID" json:"user"`
}